code-challenges.test.js

```diff
@@ -16,30 +18,102 @@
 
 // a) Create a test with an expect statement using the variable provided.
+//set up test that will describe when codedMessage gets called
+// if argument is secretCodeWord1 
+//actual result .toEqual("l4ck4d41s1c4l")
+// if argument is secretCodeWord2 
+//actual result .toEqual("g0bbl3dyg00k")
+
+describe('when codedMessage gets called',()=>{
+    it("if argument is secretCodeWord1, retrun l4ck4d41s1c4l",() =>{
+    const argument= secretCodeWord1;
+    const actualResult = codedMessage(argument)
+    expect(actualResult).toEqual("l4ck4d41s1c4l")
+    })
+    it("if argument is secretCodeWord2, retrun g0bbl3dyg00k",() =>{
+    const argument= secretCodeWord2
+    const actualResult = codedMessage(argument)
+    expect(actualResult).toEqual("g0bbl3dyg00k")
+    })
+})
+
+
+
+// // b) Create the function that makes the test pass.
+
+
+// //create a function that takes a string as an argument
+const codedMessage = (string) =>{
+// //strings.toLowerCase() and splits that string with ('') foreach of value and if value === 'a' return 4 elseif value value ==='e' return 3 elseif value === 'i' elseif value === 'o' return 0  else return value and returns that new array to a codeString.join("")
+
+    codedString = []
+    string.toLowerCase().split('').forEach((value) =>{
+        if (value === 'a')
+        return codedString.push(4);
+        else if (value ==='e')
+        return codedString.push(3);
+        else if (value === 'i')
+        return codedString.push(1);
+        else if(value === 'o')
+        return codedString.push(0);
+        else
+        return codedString.push(value)
+    })
+    return codedString.join('')
+}
+
 
 var secretCodeWord1 = "lackadaisical"
 // Expected output: "l4ck4d41s1c4l"
 var secretCodeWord2 = "gobbledygook"
 // Expected output: "g0bbl3dyg00k"
 
+// console.log(codedMessage(secretCodeWord1))
+// console.log(codedMessage(secretCodeWord2))
 
 
 
 
-// b) Create the function that makes the test pass.
-
-
 
 
 
 // --------------------2) Create a function that takes in an array and returns all the words that contain the letter a.
 
 // a) Create a test with an expect statement using the variable provided.
+//setting up test to describe when onlyAs gets called 
+// when argument is arrayofWords  
+//actual result .toEqual("Apple" "Banana" "Peach")
+
+
+describe('when onlyAs gets called',()=>{
+    it('if argument is arrayOfWords, return "Apple" "Banana" "Peach"',() =>{
+    const argument= arrayOfWords;
+    const actualResult = onlyAs(argument)
+    expect(actualResult).toEqual(["Apple", "Banana", "Peach"])
+    })
+})
+
+
+//set up a function that takes in an array of strings
+const onlyAs = (array) =>{
+    splitStrings=[]
+    array.forEach(element => {
+        //foreach elemenet in the array it sill split word then findIndex of 'Aa'
+//push(element if 'Aa' is found)
+
+       element.split('').find(value =>{
```

Comment:
nice use of `.find()`. This is a good optimization because it will stop iterating over the word as soon as the letter 'a' is identified